InstallHalyard.sh

```diff
@@ -0,0 +1,255 @@
+                                install openjdk-8-jdk.
+
+    --home_dir                  Override where user home directories reside
+                                example: /export/home vs /home
+EOF
+}
+
+function echo_status() {
+  if [ -n "$QUIET" ]; then
+    echo "$@"
+  fi
+}
+
+function process_args() {
+  while [ "$#" -gt "0" ]
+  do
+    local key="$1"
+    shift
+    case $key in
+      --repository)
+        echo "repo"
+        REPOSITORY_URL="$1"
+        shift
+        ;;
+      --dependencies_only)
+        echo "deps"
+        DEPENDENCIES_ONLY=true
+        ;;
+      --local-install)
+        echo "local"
+        DOWNLOAD=true
+        ;;
+      --quiet|-q)
+        QUIET=true
+        ;;
+      --home_dir)
+        homebase="$1"
+        if [ "$(basename $homebase)" = "spinnaker" ]; then
+          echo "stripping trailing 'spinnaker' from --home_dir=$homebase"
+          homebase=$(dirname $homebase)
+        fi
+        shift
+        ;;
+      --help|-help|-h)
+        print_usage
+        exit 13
+        ;;
+      *)
+        echo "ERROR: Unknown argument '$key'"
+        exit -1
+    esac
+  done
+}
+
+function add_apt_repositories() {
+  # Spinnaker
+  # DL Repo goes here
+  REPOSITORY_HOST=$(echo $REPOSITORY_URL | cut -d/ -f3)
+  if [[ "$REPOSITORY_HOST" == "dl.bintray.com" ]]; then
+    REPOSITORY_ORG=$(echo $REPOSITORY_URL | cut -d/ -f4)
+    # Personal repositories might not be signed, so conditionally check.
+    gpg=""
+    gpg=$(curl -s -f "https://bintray.com/user/downloadSubjectPublicKey?username=$REPOSITORY_ORG") || true
+    if [[ ! -z "$gpg" ]]; then
+      echo "$gpg" | sudo apt-key add -
+    fi
+  fi
+  echo "deb $REPOSITORY_URL $DISTRIB_CODENAME spinnaker" | sudo tee /etc/apt/sources.list.d/spinnaker-dev.list > /dev/null
+  # Java 8
+  # https://launchpad.net/~openjdk-r/+archive/ubuntu/ppa
+  sudo add-apt-repository -y ppa:openjdk-r/ppa
+  sudo apt-get update ||:
+}
+
+function install_java() {
+  if [ -z "$DOWNLOAD" ]; then
+    sudo apt-get install -y --force-yes openjdk-8-jdk
+
+    # https://bugs.launchpad.net/ubuntu/+source/ca-certificates-java/+bug/983302
+    # It seems a circular dependency was introduced on 2016-04-22 with an openjdk-8 release, where
+    # the JRE relies on the ca-certificates-java package, which itself relies on the JRE. D'oh!
+    # This causes the /etc/ssl/certs/java/cacerts file to never be generated, causing a startup
+    # failure in Clouddriver.
+    sudo dpkg --purge --force-depends ca-certificates-java
+    sudo apt-get install ca-certificates-java
+  elif [[ "x`java -version 2>&1|head -1`" != *"1.8.0"* ]]; then
+    echo "you must manually install java 8 and then rerun this script; exiting"
+    exit 13
+  fi
+}
+
+function install_halyard() {
+  package="spinnaker-halyard"
+  sudo apt-get install -y --force-yes --allow-unauthenticated $package
+  local apt_status=$?
+  if [[ $apt_status -ne 0 ]]; then
```

Comment:
I'd add some `echo`s to let people know you're going the `dpkg` route and that `apt` install failed.